chat-plugins/badges.js

```diff
@@ -53,43 +54,72 @@ exports.commands = {
-			if (!parts[1]) return this.errorReply("Invalid command. Valid commands are `/badges list`, `/badges info, badgeName`, `/badges set, user, badgeName` and `/badges take, user, badgeName`.");
-			selectedBadge = parts[1].trim();
+			if (!target) return this.errorReply("Correct command: `/badges info badgeName`.");
+			selectedBadge = target.trim();
 			if (!Db('badgeData').has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list");
 			let badgeData = Db('badgeData').get(selectedBadge);
 			this.sendReplyBox(badgeImg(badgeData[1], selectedBadge) + selectedBadge + ': ' + badgeData[0]);
-			break;
-		case 'take':
+		},
+		take: function (target, room, user) {
+			let parts = target.split(',');
+			let userid;
+			let selectedBadge;
+			let userBadges;
 			if (!this.can('lock')) return false;
-			if (parts.length !== 3) return this.errorReply("Correct command: `/badges take, user, badgeName`");
-			userid = toId(parts[1].trim());
+			if (parts.length !== 2) return this.errorReply("Correct command: `/badges take user, badgeName`");
+			userid = toId(parts[0].trim());
 			if (!Db('userBadges').has(userid)) return this.errorReply("This user doesn't have any badges.");
 			userBadges = Db('userBadges').get(userid);
-			selectedBadge = parts[2].trim();
+			selectedBadge = parts[1].trim();
 			userBadges = userBadges.filter(b => b !== selectedBadge);
 			Db('userBadges').set(toId(userid), userBadges);
+
 			this.logModCommand(user.name + " took the badge '" + selectedBadge + "' badge from " + userid + ".");
 			this.sendReply("The '" + selectedBadge + "' badge was taken from '" + userid + "'.");
-			break;
-		case 'delete':
+		},
+		delete: function (target, room, user) {
+			let selectedBadge;
 			if (!this.can('ban')) return false;
-			if (parts.length !== 2) return this.errorReply("Correct command: `/badges deleteall, badgeName`");
-			selectedBadge = parts[1].trim();
+			if (!target) return this.errorReply("Correct command: `/badges deleteall badgeName`");
+			selectedBadge = target.trim();
 			if (!Db('badgeData').has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list");
 			Db('badgeData').delete(selectedBadge);
 			let badgeUserObject = Db('userBadges').object();
 			let users = Object.keys(badgeUserObject);
 			users.forEach(u => Db('userBadges').set(u, (badgeUserObject[u].filter(b => b !== selectedBadge))));
+
 			this.sendReply("The badge with the name '" + selectedBadge + "' deleted.");
 			this.logModCommand(user.name + " removed the badge '" + selectedBadge + ".");
-			break;
-		default:
-			return this.errorReply("Invalid command. Valid commands are `/badges list`, `/badges info, badgeName`, `/badges set, user, badgeName` and `/badges take, user, badgeName`" +
-			 "`/badges create, name, description, img`.");
-		}
+		},
+		transfer: function (target, room, user) {
+			let parts = target.split(',');
+			if (!this.can('ban')) return false;
```

Comment:
I just left it as a staff command because that's how staff wanted it at the time, and it lets us control it a lot more and stop things like merging and then selling the badges.  Granted they still can merge and abuse with it as a staff command, but it will be less prevalent and we can control it more.  Also the number of people asking for badge transfers isn't high enough that handling the volume ourselves is an issue, it's just annoying to /badge take 5 times then /badge give 5 times, which was really the catalyst for staff asking for this command.